rust/connlib/tunnel/src/client.rs

```diff
@@ -1035,4 +1080,78 @@ mod tests {
     fn ignores_ip6_multicast_all_routers() {
         assert!(is_definitely_not_a_resource("ff02::2".parse().unwrap()))
     }
+
+    #[test]
+    fn dns_updated_when_dns_changes() {
+        assert!(dns_updated(
+            &["1.0.0.1".parse().unwrap()],
+            &["1.1.1.1".parse().unwrap()]
+        ))
+    }
+
+    #[test]
+    fn dns_not_updated_when_dns_remains_the_same() {
+        assert!(!dns_updated(
+            &["1.1.1.1".parse().unwrap()],
+            &["1.1.1.1".parse().unwrap()]
+        ))
+    }
+
+    #[test]
+    fn dns_updated_ignores_order() {
+        assert!(!dns_updated(
+            &["1.0.0.1".parse().unwrap(), "1.1.1.1".parse().unwrap()],
+            &["1.1.1.1".parse().unwrap(), "1.0.0.1".parse().unwrap()]
+        ))
+    }
+
+    #[test]
+    fn update_system_dns_works() {
+        let mut mock_state = client_state_fixture();
+
+        let now = Instant::now();
+        mock_state.update_system_resolvers(vec!["1.1.1.1".parse().unwrap()], now);
+        let now = now + Duration::from_millis(500);
+        mock_state.handle_timeout(now);
+
+        assert_eq!(
+            mock_state.poll_event(),
+            Some(ClientEvent::RefreshInterfance)
+        );
+    }
+
+    #[test]
+    fn update_system_dns_without_change_is_a_no_op() {
+        let mut mock_state = client_state_fixture();
+
+        let now = Instant::now();
+        mock_state.update_system_resolvers(vec!["1.1.1.1".parse().unwrap()], now);
+        let now = now + Duration::from_millis(500);
+        mock_state.handle_timeout(now);
+        mock_state.poll_event();
```

Comment:
Missing an assertion here?

Reply:
That's already asserted in the previous test, it'd be redundant